modules/islandora_scholar_embargo/includes/embargo.inc

```diff
@@ -445,24 +445,63 @@ EOQ;
 }
 
 /**
- * Retrieve all embargoed items from an object.
+ * Retrieve all embargoed items.
+ *
+ * @param bool $count
+ *   Whether or not to just return the total count.
  *
  * @return array
- *   An array containing the list of embargoed items on an object
+ *   An array containing the list of embargoed items.
  */
-function islandora_scholar_embargo_get_all_embargoed() {
+function islandora_scholar_embargo_get_all_embargoed($count = FALSE) {
   $current_embargo = <<<EOQ
 PREFIX is: <info:islandora/islandora-system:def/scholar#>
 SELECT ?obj ?date
 FROM <#ri>
 WHERE {
   ?obj is:embargo-until ?date
 }
+EOQ;
+  if ($count) {
+    $current_embargo .= <<<EOQ
 ORDER BY ?obj
 EOQ;
+  }
   $connection = islandora_get_tuque_connection();
-  $embargo_results = $connection->repository->ri->sparqlQuery($current_embargo, 'unlimited', '0');
-  return $embargo_results;
+  $ri = $connection->repository->ri;
+  return $count ? $ri->countQuery($current_embargo, 'sparql') : $ri->sparqlQuery($current_embargo, 'unlimited', '0');
+}
+
+/**
+ * Gets a subset of all embargoed items.
+ *
+ * @param string $offset
+ *   An offset ISO8601 date string to filter results before.
+ * @param int $limit
+ *   The number of items to return in the subset.
+ *
+ * @return array
+ *   A SPARQL query results array containing a subset of embargoed items,
+ *   including 'obj' as the object ID, 'expiry' as the expiry date, and 'date'
+ *   as the item's created or last modified date.
+ */
+function islandora_scholar_embargo_get_embargoed_items_subset($offset = '1900-01-01T00:00:00.000Z', $limit = 10) {
+  $current_embargo = <<<EOQ
+PREFIX is: <info:islandora/islandora-system:def/scholar#>
+SELECT DISTINCT ?obj ?expiry ?date
+FROM <#ri>
+WHERE {
+  ?obj is:embargo-until ?expiry ;
+  { ?obj <fedora-model:createdDate> ?date } UNION
+  { ?obj <fedora-view:lastModifiedDate> ?date ;
+         <fedora-view:disseminationType> ?dsid
+  }
+  FILTER(?date > "$offset"^^<http://www.w3.org/2001/XMLSchema#dateTime>)
```

Comment:
Could miss items... If two items happen to have the same date. Would have to be something like "date greater or (date equal and something to break the tie)"... Usually using the PID...
```sparql
PREFIX is: <info:islandora/islandora-system:def/scholar#>
SELECT DISTINCT ?obj ?expiry ?date
FROM <#ri>
WHERE {
  ?obj is:embargo-until ?expiry ;
  { ?obj <fedora-model:createdDate> ?date } UNION
  { ?obj <fedora-view:lastModifiedDate> ?date ;
         <fedora-view:disseminationType> ?dsid
  }
  FILTER(?date > "$offset"^^<http://www.w3.org/2001/XMLSchema#dateTime> || (?date = "$offset"^^<http://www.w3.org/2001/XMLSchema#dateTime> && ?obj > "$the_last_one"))
}
ORDER BY ASC(?date), ASC(?obj)
```